pmm-app/src/shared/components/Elements/Modal/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC, useEffect } from 'react';
+
+import { IconButton, useTheme } from '@grafana/ui';
+import { getStyles } from './Modal.styles';
+
+interface ModalWindow {
+  onClose: () => void;
+  closeOnClickaway?: boolean;
+  closeOnEscape?: boolean;
+  isVisible: boolean;
+  title: FC | string;
+}
+
+
+export const Modal: FC<ModalWindow> = (props) => {
+  const {
+    isVisible, children, title, onClose, closeOnClickaway = true, closeOnEscape = true
+  } = props;
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  if (closeOnEscape) {
+    useEffect(() => {
```

Comment:
How about this rule from https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level?
`Don’t call Hooks inside loops, conditions, or nested functions`. 

I wonder, doesn't eslint complain about that?

Reply:
Good point, fixed